golang/cosmos/x/vbank/vbank_test.go

```diff
@@ -82,7 +82,11 @@ func decodeBalances(encoded []byte) (balances, uint64, error) {
 		return nil, 0, fmt.Errorf("bad balance update type: %s", balanceUpdate.Type)
 	}
 	b := newBalances()
-	for _, u := range balanceUpdate.Updated {
+	fmt.Printf("updated balances %v\n", balanceUpdate.Updated)
```

Comment:
The non-determinism of update order is what led to the following commit in the PR that created these tests:  https://github.com/Agoric/agoric-sdk/pull/3210/commits/854fe21b192550642d1314698b9610980c593560

In the long run, that commit should be undone and this file should switch from `want` values in the the map-of-maps representation - elegant as it is - to the `vbankBalanceUpdate` structure, or perhaps the raw encoded JSON - depending on whether differences in encoded JSON, e.g. different whitespace, would trigger a nondeterminism failure.

That could possibly be delayed to a follow-on PR. If so, then for now please abstract this plus the Type field check above into a `balanceUpdateValid(vbu vbankBalanceUpdate) bool` utility function, and file an issue for going back to determinism-sensitive `want` values.

Reply:
I introduced `validateBalanceUpdate`, which returns an `error` if there's a problem, `nil` otherwise.

I'm not sure I want determinism-sensitive `want` values.  I like the fact that the test creates random private/pubkeys, and it would make the test less clear to write dynamic `want` values (since their update order depends on the lexicographic ordering of the pubkey addresses).